packages/ckeditor5-core/src/multicommand.js

```diff
@@ -56,23 +58,44 @@ export default class MultiCommand extends Command {
 	}
 
 	/**
-	 * Executes the first of it registered child commands.
+	 * Executes the first enabled command which has the highest priority of all registered child commands.
 	 *
 	 * @returns {*} The value returned by the {@link module:core/command~Command#execute `command.execute()`}.
 	 */
 	execute( ...args ) {
 		const command = this._getFirstEnabledCommand();
 
-		return command != null && command.execute( args );
+		return !!command && command.execute( args );
+	}
+
+	/**
+	 * Inserts command definition at correct index by priority so registered commands are always sorted from lowest priority to highest
+	 *
+	 * @param {Object} newCommandDefinition Object with `command` and `priority` properties
+	 * Object
+	 * @returns {undefined}
+	 * @private
+	 */
+	_insertCommandDefinitionByPriority( newCommandDefinition ) {
+		for ( let i = 0; i <= this._childCommandsDefinitons.length; i++ ) {
+			const registeredCommand = this._childCommandsDefinitons[ i ];
+
+			if ( !registeredCommand || priorities.get( registeredCommand.priority ) >= priorities.get( newCommandDefinition.priority ) ) {
+				this._childCommandsDefinitons.splice( i, 0, newCommandDefinition );
+
+				break;
+			}
+		}
```

Comment:
This is almost exactly the same as https://github.com/ckeditor/ckeditor5/blob/af03d908ff9d1e37938fe73f6a5d42fff3eb33c3/packages/ckeditor5-utils/src/emittermixin.js#L291-L316 so maybe we could extract it to some util in the utils package? It would accept a list, an item to insert, and a priority (to insert the item at the correct index).